src/gafaelfawr/verify.py

```diff
@@ -337,12 +342,12 @@ async def _get_keys(self, issuer_url: str) -> List[Dict[str, str]]:
             raise FetchKeysException(msg)
 
         body = await r.json()
-        if "keys" not in body:
+        try:
+            return body["keys"]
+        except Exception:
```

Comment:
I think I liked the pattern before with looking at the key rather than catching exception, but it's fine either way (not sure if I'm missing some subtle detail)

Reply:
This is super, super-fiddly and probably won't come up in practice, but I made this change because I realized that if upstream returned a JSON list with `keys` as one of the members of the list, it would pass the if statement and then produce an uncaught exception. This way of writing it ensures we catch the exception so that we can report an error message to the user.

Reply:
Sounds good, that is hilariously subtle.